### PR TITLE
fix(core/modal): figma - code mismatch

### PR DESCRIPTION
**Pull request by matthiashader**
## 💡 What is the current behavior?
- Modal closing X not aligned with buttons
- Icon size 24px
- Indent of text not aligned with header indent
- Modal padding: 14px (according to ticket, but was already 16px)
- Modal content padding: 10px

GitHub Issue Number: #-, [IX-1454]

## 🆕 What is the new behavior?
- Modal closing X aligned with buttons
- Icon size 32px
- Indent of text aligned with header indent
- Modal padding: 16px
-  Modal content padding: 8px

## 🏁 Checklist

A pull request can only be merged if all of these conditions are met (where applicable):

- [ ] 🦮 Accessibility (a11y) features were implemented
- [ ] 🗺️ Internationalization (i18n) - no hard coded strings
- [ ] 📲 Responsiveness - components handle viewport changes and content overflow gracefully
- [ ] 📄 Documentation was reviewed/updated (`pnpm run docs`)
- [ ] 🧪 Unit tests were added/updated and pass (`pnpm test`)
- [ ] 📸 Visual regression tests were added/updated and pass ([Guide](https://github.com/siemens/ix/blob/main/CONTRIBUTING.md#visual-regression-testing))
- [ ] 🧐 Static code analysis passes (`pnpm lint`)
- [ ] 🏗️ Successful compilation (`pnpm build`, changes pushed)

### CODE REVIEW

File: packages/core/src/components/modal-header/modal-header.scss [hunk 1]
```diff
@@ -11,13 +11,12 @@
 :host {
   @include ix-component;
   display: flex;
-  padding: 0.5rem;
+  padding: 0.5rem 0 0.5rem 0.5rem;
   align-items: center;
   align-self: stretch;
 
   .modal-close {
     margin-left: auto;
```
Comment (nuke-ellington) [inline packages/core/src/components/modal-header/modal-header.scss:19]:
```suggestion
    margin-inline-start: 1rem;
```

**Review by nuke-ellington — CHANGES_REQUESTED** [review]

File: packages/core/src/components/modal-header/modal-header.scss (hunk 1 above)
Comment (matthiashader) [reply packages/core/src/components/modal-header/modal-header.scss:19]: Why do we need this? If i apply this change the icon would be placed next to the text?
image

**Review by AndreasBerliner — DISMISSED** [review]

**Review by nuke-ellington — APPROVED** [review]

**Review by danielleroux — CHANGES_REQUESTED** [review]
Strict errors not reduces in modal-header component

File: packages/core/src/components/modal-header/modal-header.tsx [hunk 2]
```diff
@@ -54,22 +54,24 @@ export class ModalHeader {
...
 
   componentDidLoad() {
     this.parentDialog = closestPassShadow(
       this.hostElement,
       'ix-modal'
     ) as HTMLIxModalElement;
-    this.onIconChange(this.icon);
+    if (this.icon) {
+      this.onIconChange(this.icon);
+    }
```
Comment (danielleroux) [inline packages/core/src/components/modal-header/modal-header.tsx:74]: This additional check change the  behaviour of the component.

**Review by danielleroux — CHANGES_REQUESTED** [review]

**Review by danielleroux — APPROVED** [review]

**Pull request merged**